GitUI/CommandsDialogs/SearchControl.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Windows.Forms;
+using GitCommands;
+
+namespace GitUI.CommandsDialogs
+{
+    public partial class SearchControl<T> : UserControl, IDisposable where T : class
+    {
+        private readonly Func<string, IList<T>> getCandidates;
+        private readonly Action<Size> _onSizeChanged;
+        private AsyncLoader backgroundLoader = new AsyncLoader();
```

Comment:
Can we use the standard `async/await` constructs instead of the `AsyncLoader`?
There is currently work championed by @drewnoakes to remove `AsyncLoader` completely

Reply:
Has the pattern for `async/await` been finalized? I have noticed the `AsyncLoader` is quite complex, what is the equivalent code of using `async/await`?